rust/src/rfb/rfb.rs

```diff
@@ -205,7 +215,10 @@ impl RFBState {
                             );
                         }
                         Err(_) => {
-                            return AppLayerResult::err();
+                            // We even failed to parse the protocol version.
+                            // Continue the flow but stop trying to parse the protocol.
```

Comment:
What sense does it make to try to continue app-layer parsing / reassembly here ?
Why not return error ?

Reply:
So, I understand that for unknown values for rust enumerations, we do not want to error, but maybe get in a state where we can no longer analyze...

But if the first pass of parsing the protocol version is unrecognized, I think this should still be an error

Reply:
YEah, you are probably right. My approach here was to never cause RFB to be interrupted for parsing reasons, but maybe this is a bit too lenient. I'll make this an error.